client.go

```diff
@@ -110,6 +152,75 @@ func (c *Client) SetUserAgent(ua string) *Client {
 	return c
 }
 
+type RequestParams struct {
+	Body     map[string]any
+	Response any
+}
+
+// Generic helper to execute HTTP requests using the
+//
+//nolint:unused
+func (c *HTTPClient) doRequest(ctx context.Context, method, url string, params RequestParams, mutators ...func(req *http.Request) error) error {
+	// Create a new HTTP request
+	var bodyReader io.Reader
+	if params.Body != nil {
+		buf := new(bytes.Buffer)
+		if err := json.NewEncoder(buf).Encode(params.Body); err != nil {
+			return fmt.Errorf("failed to encode body: %w", err)
+		}
+		bodyReader = buf
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, url, bodyReader)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Set default headers
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	if c.userAgent != "" {
+		req.Header.Set("User-Agent", c.userAgent)
+	}
+
+	// Apply mutators
+	for _, mutate := range mutators {
+		if err := mutate(req); err != nil {
+			return fmt.Errorf("failed to mutate request: %w", err)
+		}
+	}
+
+	// Send the request
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Check for HTTP errors
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		var errorDetails string
+		if resp.Body != nil {
+			bodyBytes, err := io.ReadAll(resp.Body)
+			if err != nil {
+				errorDetails = fmt.Sprintf("failed to read response body: %v", err)
+			} else {
+				errorDetails = string(bodyBytes)
+			}
+		}
+		return fmt.Errorf("received non-2xx status code: %d, details: %s", resp.StatusCode, errorDetails)
```

Comment:
To keep errors unified we might want to build a [`linodego.Error`](https://github.com/linode/linodego/blob/main/errors.go#L24) object here, which may involve creating an equivalent to `coupleAPIErrors` but takes a `*http.Response` rather than a `*resty.Response`.

Reply:
Thanks for the tip! I will implement this.

Reply:
Fixed!